From a97a61207bd61d04c7aa78a171c36e3bfc25b05f Mon Sep 17 00:00:00 2001
From: Kevin Hao <kexin.hao@windriver.com>
Date: Tue, 14 Apr 2015 14:09:05 +0800
Subject: [PATCH] powerpc: dts: add nor flash partitions for p2041rdb and
 p3041ds

Both these boards have a 128M nor flash and create the same default
partitions layout for them.

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/powerpc/boot/dts/p2041rdb.dts |   59 +++++++++++++++++++++++++++++++++++
 arch/powerpc/boot/dts/p3041ds.dts  |   60 ++++++++++++++++++++++++++++++++++++
 2 files changed, 119 insertions(+), 0 deletions(-)

diff --git a/arch/powerpc/boot/dts/p2041rdb.dts b/arch/powerpc/boot/dts/p2041rdb.dts
index 1df2728..b3be1bb 100644
--- a/arch/powerpc/boot/dts/p2041rdb.dts
+++ b/arch/powerpc/boot/dts/p2041rdb.dts
@@ -268,10 +268,69 @@
 			  1 0 0xf 0xffa00000 0x00040000>;
 
 		flash@0,0 {
+			#size-cells = <1>;
+			#address-cells = <1>;
 			compatible = "cfi-flash";
 			reg = <0 0 0x08000000>;
 			bank-width = <2>;
 			device-width = <2>;
+
+			partition@0 {
+				label = "rcw";
+				reg = <0x0 0x20000>;
+				read-only;
+			};
+			partition@20000 {
+				label = "kernel";
+				reg = <0x20000 0x700000>;
+			};
+			partition@720000 {
+				label = "dtb";
+				reg = <0x720000 0x20000>;
+			};
+			partition@740000 {
+				label = "fs1";
+				reg = <0x740000 0x37c0000>;
+			};
+			partition@3f00000 {
+				label = "alt-fman-firmware";
+				reg = <0x3f00000 0x20000>;
+				read-only;
+			};
+			partition@3f20000 {
+				label = "alt-u-boot-env";
+				reg = <0x3f20000 0x20000>;
+				read-only;
+			};
+			partition@3f40000 {
+				label = "alt-u-boot";
+				reg = <0x3f40000 0xc0000>;
+				read-only;
+			};
+			partition@4000000 {
+				label = "alt-rcw";
+				reg = <0x4000000 0x20000>;
+				read-only;
+			};
+			partition@4020000 {
+				label = "fs2";
+				reg = <0x4020000  0x3ee0000>;
+			};
+			partition@7f00000 {
+				label = "fman-firmware";
+				reg = <0x7f00000 0x20000>;
+				read-only;
+			};
+			partition@7f20000 {
+				label = "u-boot env";
+				reg = <0x7f20000  0x20000>;
+				read-only;
+			};
+			partition@7f40000 {
+				label = "u-boot";
+				reg = <0x7f40000 0xc0000>;
+				read-only;
+			};
 		};
 
 		nand@1,0 {
diff --git a/arch/powerpc/boot/dts/p3041ds.dts b/arch/powerpc/boot/dts/p3041ds.dts
index 2ec61f8..fe9a1c4 100644
--- a/arch/powerpc/boot/dts/p3041ds.dts
+++ b/arch/powerpc/boot/dts/p3041ds.dts
@@ -277,10 +277,70 @@
 			  3 0 0xf 0xffdf0000 0x00008000>;
 
 		flash@0,0 {
+			#size-cells = <1>;
+			#address-cells = <1>;
 			compatible = "cfi-flash";
 			reg = <0 0 0x08000000>;
 			bank-width = <2>;
 			device-width = <2>;
+
+			partition@0 {
+				label = "rcw";
+				reg = <0x0 0x20000>;
+				read-only;
+			};
+			partition@20000 {
+				label = "kernel";
+				reg = <0x20000 0x700000>;
+			};
+			partition@720000 {
+				label = "dtb";
+				reg = <0x720000 0x20000>;
+			};
+			partition@740000 {
+				label = "fs1";
+				reg = <0x740000 0x37c0000>;
+			};
+			partition@3f00000 {
+				label = "alt-fman-firmware";
+				reg = <0x3f00000 0x20000>;
+				read-only;
+			};
+			partition@3f20000 {
+				label = "alt-u-boot-env";
+				reg = <0x3f20000 0x20000>;
+				read-only;
+			};
+			partition@3f40000 {
+				label = "alt-u-boot";
+				reg = <0x3f40000 0xc0000>;
+				read-only;
+			};
+			partition@4000000 {
+				label = "alt-rcw";
+				reg = <0x4000000 0x20000>;
+				read-only;
+			};
+			partition@4020000 {
+				label = "fs2";
+				reg = <0x4020000  0x3ee0000>;
+			};
+			partition@7f00000 {
+				label = "fman-firmware";
+				reg = <0x7f00000 0x20000>;
+				read-only;
+			};
+			partition@7f20000 {
+				label = "u-boot env";
+				reg = <0x7f20000  0x20000>;
+				read-only;
+			};
+			partition@7f40000 {
+				label = "u-boot";
+				reg = <0x7f40000 0xc0000>;
+				read-only;
+			};
+
 		};
 
 		nand@2,0 {
-- 
1.7.5.4

